Thumbnail/ConsumerThumbnail.php

```diff
@@ -23,16 +25,19 @@ class ConsumerThumbnail implements ThumbnailInterface
 
     protected $backend;
 
+    protected $dispatcher;
+
     /**
      * @param string                                              $id
      * @param ThumbnailInterface                                  $thumbnail
      * @param \Sonata\NotificationBundle\Backend\BackendInterface $backend
      */
-    public function __construct($id, ThumbnailInterface $thumbnail, BackendInterface $backend)
+    public function __construct($id, ThumbnailInterface $thumbnail, BackendInterface $backend, EventDispatcherInterface $dispatcher)
```

Comment:
Nope. Just make the class working both with old and new constructors parameters.

In this case, you can make this argument optional and trigger a deprecation error if it's not specified.

Please refer to Symfony methods.